src/slot.rs

```diff
@@ -2112,15 +2092,21 @@ mod test {
                 connector.clone(),
             );
 
-            // Let the connections fill up
-            set.monitor()
-                .wait_for(|state| matches!(state, SetState::Online { .. }))
-                .await
-                .unwrap();
-
             // Claim some of the handles
             let mut handles = vec![];
             for _ in 0..config.claimed {
+                // Claim connections as they're available
```

Comment:
This is a change to a test that *shouldn't* be related to this PR, but it was flake I noticed as a result of timing differences by moving things around.

Basically: If we try to claim a handle from a `Set`, and we wait for it to be "Online", that does not necessarily mean "unclaimed slots exist". That might mean: "1 / 4 slots are connected, 3 are getting ready, and you already claimed the one we thought was good to go".

To fix: wait for claims to be accessible before claiming.

(In the more holistic view of this code, this is normally be the responsibility of the pool, to keep a queue of requests and service them as slot sets have more unclaimed slots)